### DISCUSSION TITLE
Recording streams with VP8 codec

### DISCUSSION THREAD

**Discussion by filipekav** [bluenviron/mediamtx#3625] [General]
Hello everyone.
I have been using mediamtx to publish and read webrtc streams via WHIP and WHEP. I was thinking about implementing recording of my streams.
But since I use the VP8 codec I noticed that it is not compatible with recording.
Can I expect that someday in the future this recording compatibility using VP8 will be implemented? Or is there some incompatibility that makes it impossible to record using this codec?

**Comment by aler9** [comment]
Hello, the reason why recording VP8 track is not supported is that currently there's no specification that dictates how to write VP8 tracks inside MP4 or MPEG-TS container formats.

Even if we write down a specification, all media players are currently preventing VP8 from being muxed/demuxed with MP4, for instance FFmpeg:
https://github.com/FFmpeg/FFmpeg/blob/654bd47716c4f36719fb0f3f7fd8386d5ed0b916/libavformat/movenc.c#L7941-L7946

The only container that supports VP8 is Matroska / MKV. Adding support for an additional container inside the server is not worth the effort since the codec has a low usage share.

Switch to AV1 or VP9.

**Reply by filipekav** [reply]
I understand. Thanks for the explanation.

[a comment by an automated account is not shown]
